### ISSUE TITLE
MaxDegreeOfParallelism missing in node sdk

### ISSUE THREAD

**Issue by cgurjar** [Azure/azure-documentdb-node#149]
I have a partitioned collection. I have to execute a query using node js that actually spans across partitions. using options i set EnableCrossPartitionQuery to true, but i see in node js sdk the MaxDegreeOfParallelism property is missing (I saw this in the .net sdk). i want to query all my partitions in parallel. Any help. I am writing the code using nodejs

**Comment by moderakh** [comment]
@cgurjar

If your query is top/orderby parallel behaviour just works. All you need to do is to enable EnableCrossPartitionQuery.
You can see this as a sample:
https://github.com/Azure/azure-documentdb-node/blob/master/source/test/orderbyTests.js#L401-L416
(nodejs lang by design is single threaded with async-io and we don't need any MaxDegreeOfParallelism)

But in case of non top/orderby a specific http header (i.e., "x-ms-documentdb-query-parallelizecrosspartitionquery") needs to be explicitly set to enable the parallelized behaviour. And that's currently missing. We will fix this in the next few days.

**Comment by moderakh** [comment]
@cgurjar let met correct my above comment. We decided to add support for `MaxDegreeOfParallelism` similar to .Net SDK in nodejs sdk. :-)

We implemented this, and it will be in the next release.

**Comment by cgurjar** [comment]
Hmm. Thanks. But to be honest with you for me it is a bit late., I really repent it not available when our POCs was going on. I was pushing for webservice implemented in  NodeJs (as our web service would be just talking to the documentDB - read pure IO operation). I had seen lots of success with NodeJS, especially for IO work, bu the nodeJS sdk for document DB when i tested with all possible options at that time (MaxDegreeOfParallelism was not there), it basically took much much more time than the implementation of dotnet sdk for Document DB. 
And this impacted our decision to go with DotNet (which i was not much in favour of. I know async await sort of allows non blocking io, but i felt NodeJS was more fit as it was designed for io
On another note, why did it take so long for this to be available in NodeJs sdk while it was available in dotnet sdk long time back. Is it because of low priority given to NodeJS?

**Comment by mingaliu** [comment]
@cgurjar Due to the resource constraint, we implemented this feature later than we like to.  Currently, we are monitoring our customer's requirement/ask more closely and plan our release accordingly.

**Comment by cgurjar** [comment]
@mingaliu Hmm. When using NodeJS, we usually write stuff to extract max asynchrony. And not having to send queries in parallel to the documentDb killed my proposal of NodeJs. In future please dont push those features that matter to performance. Because it makes your customer make choices that they would not normally make

**Comment by moderakh** [comment]
addressed in 1.11.0
